alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/boundary/interactions/InteractionManager.kt

```diff
@@ -0,0 +1,492 @@
+                    selection -= node.key
+                } else {
+                    selection[node.key] = node.value
+                }
+            }
+        }
+        selectionCandidatesMutex.acquireUninterruptibly()
+        selectionHelper.close()
+        selectionCandidates.clear()
+        selectionCandidatesMutex.release()
+        feedback = feedback - Interaction.SELECTION_BOX
+        repaint()
+        event.consume()
+    }
+
+    /**
+     * Called when a select gesture is canceled.
+     */
+    private fun onSelectCanceled(event: MouseEvent) {
+        selectionHelper.close()
+        feedback = feedback - Interaction.SELECTION_BOX
+        feedback = feedback - Interaction.HIGHLIGHT_CANDIDATE
+        repaint()
+        event.consume()
+    }
+
+    /**
+     * Returns a lambda which paints a highlight when called.
+     * @param position the position of the highlight
+     * @param paint the colour of the highlight
+     */
+    private fun paintHighlight(position: P, paint: Paint): () -> Unit = {
+        highlighter.graphicsContext2D.let { graphics ->
+            graphics.fill = paint
+            wormhole.getViewPoint(position).let {
+                graphics.fillOval(
+                    it.x - highlightSize / 2,
+                    it.y - highlightSize / 2,
+                    highlightSize,
+                    highlightSize
+                )
+            }
+        }
+    }
+
+    /**
+     * Grabs all the nodes in the selection box and adds them to the selection candidates.
+     */
+    private fun addNodesToSelectionCandidates() {
+        selectionCandidatesMutex.acquireUninterruptibly()
+        selectionHelper.boxSelection(nodes, wormhole).let {
+            if (it.entries != selectionCandidates.entries) {
+                selectionCandidates.clear()
+                selectionCandidates += it
+            }
+        }
+        selectionCandidatesMutex.release()
+    }
+
+    /**
+     * Clears and then paints every currently active feedback.
+     */
+    fun repaint() {
+        Platform.runLater {
+            selector.clear()
+            highlighter.clear()
+            feedback.values.forEach { it.forEach { it() } }
+        }
+    }
+
+    /**
+     * Sets the wormhole.
+     */
+    fun setWormhole(wormhole: Wormhole2D<P>) {
+        this.wormhole = wormhole
+    }
+
+    /**
+     * Sets the zoom manager.
+     */
+    fun setZoomManager(zoomManager: ZoomManager) {
+        this.zoomManager = zoomManager
+    }
+
+    /**
+     * To be called by the [AbstractFXDisplay] whenever a step occurs.
+     */
+    fun simulationStep() {
+        addNodesToSelectionCandidates()
+    }
+
+    companion object {
+        /**
+         * The size (radius) of the highlights.
+         */
+        const val highlightSize = 10.0
+        /**
+         * Empiric zoom scale value.
+         */
+        private const val ZOOM_SCALE = 40.0
+    }
+
+    private class Colors private constructor() {
```

Comment:
why not a private `object` instead of a class with a `companion`?

Reply:
No reason, fixed